src/BackgroundTask/MonitorCssTransientCaching.php

```diff
@@ -164,9 +237,39 @@ public function query_css_transient_count() {
 	 * @param string $old_version Old version.
 	 */
 	public function handle_plugin_update( $old_version ) {
-		// Reset the disabling of the CSS caching subsystem when updating from versions 1.5.0 or 1.5.1.
-		if ( version_compare( $old_version, '1.5.0', '>=' ) && version_compare( $old_version, '1.5.2', '<' ) ) {
-			AMP_Options_Manager::update_option( Option::DISABLE_CSS_TRANSIENT_CACHING, false );
+		// Note: We cannot use the is_css_transient_caching_disabled method because we need to get the underlying stored value.
+		$disabled = AMP_Options_Manager::get_option( Option::DISABLE_CSS_TRANSIENT_CACHING, false );
+		if ( empty( $disabled ) ) {
+			return;
+		}
+
+		// Obtain the version of WordPress and Gutenberg at which time the functionality was disabled, if available.
+		$wp_version        = isset( $disabled[ self::WP_VERSION ] ) ? $disabled[ self::WP_VERSION ] : null;
+		$gutenberg_version = isset( $disabled[ self::GUTENBERG_VERSION ] ) ? $disabled[ self::GUTENBERG_VERSION ] : null;
+
+		if (
+			// Reset the disabling of the CSS caching subsystem when updating from versions 1.5.0 or 1.5.1.
+			(
+				version_compare( $old_version, '1.5.0', '>=' )
+				&&
+				version_compare( $old_version, '1.5.2', '<' )
+			)
+			||
+			// Reset when it was disabled prior to the versions of WP/Gutenberg being captured,
+			// or if the captured versions were affected at the time of disabling.
+			(
+				version_compare( strtok( $old_version, '-' ), '2.2.2', '<' )
+				&&
+				(
+					! is_array( $disabled )
+					||
+					$this->block_uniqid_transformer->is_affected_gutenberg_version( $gutenberg_version )
+					||
+					$this->block_uniqid_transformer->is_affected_wordpress_version( $wp_version )
```

Comment:
I realize that these conditions probably will never happen because when the `$old_version` is older than 2.2.2, the `$disabled` value will always be `true` here.